task11/ShipmentDecorator.ts

```diff
@@ -0,0 +1,62 @@
+import { Shipment, IShipment, ShipmentType } from "./Shipment";
+import { Shipper } from "./Shipper";
+import { State } from "./State";
+
+class ShipmentDecorator implements IShipment {
+    protected wrappee: Shipment;
+    
+    // how to hide these props? currently only need in the Shipment class
+    state: State;
+    shipper: Shipper;
+    shipmentType: ShipmentType;
+
+    constructor(shipment: Shipment) {
+        this.wrappee = shipment;
+    }
+    
+    public addMark(mark?: string): void {
+        this.wrappee.addMark(mark);
+    }
+
+    // how to hide these func? currently only need in the Shipment class
+    public ship(): string {
+        return this.wrappee.ship();
+    }
```

Comment:
This function should be a little bit change
You need to get results from this.wrapped.ship() and add to this string all needed marks
Smth like: 
`return `${this.wrappee.ship() \n ${marks}};`
Please, don't forget that marks are array, so you need to make from it needed string :)